src/pages/admin/MiradorForm.js

```diff
@@ -0,0 +1,503 @@
+            window: {
+              allowClose: false,
+              allowFullscreen: true,
+              allowMaximize: false,
+              allowTopMenuButton: mirador.window.allowTopMenuButton || true,
+              allowWindowSideBar: mirador.window.allowWindowSideBar || true,
+              sideBarPanel: mirador.window.sideBarPanel || "info",
+              defaultView: "single",
+              sideBarOpen: mirador.window.sideBarOpen || false,
+              panels: {
+                info: true,
+                attribution: true,
+                canvas: false,
+                annotations: true,
+                search: false,
+                layers: false
+              }
+            },
+            windows: [
+              {
+                manifestId: ""
+              }
+            ],
+            thumbnailNavigation: {
+              defaultPosition:
+                mirador.thumbnailNavigation.defaultPosition || "far-bottom",
+              displaySettings:
+                mirador.thumbnailNavigation.displaySettings || true,
+              height: mirador.thumbnailNavigation.height || 130,
+              width: mirador.thumbnailNavigation.width || 100
+            },
+            workspace: {
+              draggingEnabled: false,
+              allowNewWindows: false,
+              isWorkspaceAddVisible: false,
+              showZoomControls: true,
+              type: "mosaic"
+            },
+            workspaceControlPanel: {
+              enabled: false
+            }
+          };
+        } catch (error) {
+          console.error(error);
+        }
+        this.setState({
+          formState: siteInfo,
+          prevFormState: siteInfo,
+          site: site
+        });
+      } else {
+        let siteInfo = {
+          selectedTheme: "light",
+          language: "en",
+          id: "mirador_viewer",
+          window: {
+            allowClose: false,
+            allowFullscreen: true,
+            allowMaximize: false,
+            allowTopMenuButton: true,
+            allowWindowSideBar: true,
+            sideBarPanel: "info",
+            defaultView: "single",
+            sideBarOpen: false,
+            panels: {
+              info: true,
+              attribution: true,
+              canvas: false,
+              annotations: true,
+              search: false,
+              layers: false
+            }
+          },
+          windows: [
+            {
+              manifestId: ""
+            }
+          ],
+          thumbnailNavigation: {
+            defaultPosition: "far-bottom",
+            displaySettings: true,
+            height: 130,
+            width: 100
+          },
+          workspace: {
+            draggingEnabled: false,
+            allowNewWindows: false,
+            isWorkspaceAddVisible: false,
+            showZoomControls: true,
+            type: "mosaic"
+          },
+          workspaceControlPanel: {
+            enabled: false
+          }
```

Comment:
Code block 2

Reply:
It looks like there is a repeat code in `Code block 1` and `Code block 2` and could be merged.

Reply:
I think I need to keep the else clause because initially the digital libraries do not have any config data for miradorOptions, so this supplies the default values if miradorOptions data does not already exist. 
![image](https://user-images.githubusercontent.com/38388061/105233811-58bcda80-5b38-11eb-977d-be66dcaf7d34.png)

Reply:
line 27-77 and line 79-127 still are repeat codes, please refactor it.

Reply:
This is resolved now.

Reply:
@yinlinchen Just in case the notification got lost in the mail, I'm sending a reminder that this PR has been refactored and is good to go.